.ci/scripts/clang-format.sh

```diff
@@ -20,9 +20,11 @@ echo "Starting format check..."
 
 RETURN_CODE=0
 
+CLANG_FORMAT_EXE=${CLANG_FORMAT_EXE:-clang-format-14}
```

Comment:
Is it necessary to set specific version here as default? Because we generally do not update this file, the default clang-format option would be used which provides some flexibility

Reply:
I observed this issue since the current CI uses `clang-format-14`, our systems had newer versions. The newer formatting was not compatible with `clang-format-14` and hence failing the CI check. This allows to use the default version based on CI version.